src/main/kotlin/uk/gov/justice/digital/hmpps/locationsinsideprison/service/LocationService.kt

```diff
@@ -1176,6 +1178,33 @@ class LocationService(
     }.sortedBy { it.pathHierarchy }
   }
 
+  fun getCellAttributes(id: UUID): List<CellAttributes> {
+    val cell = cellLocationRepository.findById(id).getOrNull()
+      ?: throw LocationNotFoundException(id.toString())
+
+    val specialistCellTypes = cell.specialistCellTypes
+      .map {
+        CellAttributes(
+          it.specialistCellType,
+          it.specialistCellType.description,
+        )
+      }
+
+    val residentialAttributes = cell.attributes.filter { it.attributeType == ResidentialAttributeType.LOCATION_ATTRIBUTE }
+      .map {
+        CellAttributes(
+          it.attributeValue,
+          it.attributeValue.description,
+        )
+      }
+
+    return if (activePrisonConfig.isActivePrison(cell.prisonId)) {
+      specialistCellTypes
```

Comment:
Potentially you can add the mapping inside the if statement as you don't need to map both only the one you need